File: greedy/leetcode_605/solution.py
from typing import List

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0
        padded_flowerbed = [0] + flowerbed + [0]
        
        for i in range(1, len(padded_flowerbed) - 1):
            if padded_flowerbed[i] == 0 and \
               padded_flowerbed[i - 1] == 0 and \
               padded_flowerbed[i + 1] == 0:
                
                padded_flowerbed[i] = 1 
                count += 1              

                if count >= n:
                    return True
        
        return count >= n

File: greedy/leetcode_605/solution.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
    let result = 0;
    const paddedFlowerbed = [0, ...flowerbed, 0];
    const bedLength = paddedFlowerbed.length;

    for (let i=0; i<bedLength - 1; i++) {
        if (paddedFlowerbed[i] === 0 && 
            paddedFlowerbed[i - 1] === 0 && 
            paddedFlowerbed[i + 1] === 0) {
            
            paddedFlowerbed[i] = 1;
            result++;

            if (result >= n) {
                return true;
            }
        }
    }

    return result >= n;
};